specs/GetFavicon.md

```diff
@@ -0,0 +1,172 @@
+/// FaviconUri property and GetFavicon method to obtain the favicon 
+/// data. The second argument is always null.
+/// For more information see `add_FaviconChanged`.
+[uuid(2913DA94-833D-4DE0-8DCA-900FC524A1A4), object, pointer_default(unique)]
+interface ICoreWebView2FaviconChangedEventHandler : IUnknown {
+  /// Called to notify the favicon changed.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+/// This interface is a handler for the completion of the population of
+/// `imageStream`.
+/// `errorCode` returns S_OK if the API succeeded.
+/// The image is returned in the `faviconStream` object. If there is no image
+/// then no data would be copied into the imageStream.
+/// For more details, see the `GetFavicon` API.
+[uuid(A2508329-7DA8-49D7-8C05-FA125E4AEE8D), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalGetFaviconCompletedHandler : IUnknown {
+  /// Called to notify the favicon has been retrieved.
+  HRESULT Invoke(
+    [in] HRESULT errorCode,
+    [in] IStream* faviconStream);
+}
+
+/// This is the ICoreWebView2 Experimental Favicon interface.
+[uuid(DC838C64-F64B-4DC7-98EC-0992108E2157), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+    /// Add an event handler for the `FaviconChanged` event.
+    /// The `FaviconChanged` event is raised when the 
+    /// [favicon](https://developer.mozilla.org/en-US/docs/Glossary/Favicon)
+    /// had a different URL then the previous URL.
+    /// The FaviconChanged event will be raised for first navigating to a new 
+    /// document, whether or not a document declares a Favicon in HTML if the
+    /// favicon is different from the previous fav icon. The event will 
+    /// be raised again if a favicon is declared in its HTML or has script 
+    /// to set its favicon. The favicon information can then be retrieved with 
+    /// `GetFavicon` and `FaviconUri`.
+    HRESULT add_FaviconChanged(
+        [in] ICoreWebViewFaviconChangedEventHandler* eventHandler,
+        [out] EventRegistrationToken* token);
+
+    /// Remove the event handler for `FaviconChanged` event.
+    HRESULT remove_FaviconChanged(
+        [in] EventRegistrationToken token);
+
+    /// Get the current Uri of the favicon as a string.
+    /// If the value is null, then the return value is `E_POINTER`, otherwise it is `S_OK`.
+    /// If a page has no favicon then the value is an empty string.
+    [propget] HRESULT FaviconUri([out, retval] LPWSTR* value);
+
+    /// Async function for getting the actual image data of the favicon.
+    /// If the `imageStream` is null, the `HRESULT` will be `E_POINTER`, otherwise
+    /// it is `S_OK`.
+    /// The image is copied to the `imageStream` object. If there is no image then
+    /// no data would be copied into the imageStream.
+    /// In either scenario the `completedHandler` is executed at the end of the operation.
+    HRESULT GetFavicon(
```

Comment:
does the cpp projection actually drop the 'Async' suffix?

Reply:
That is correct, the C++ does not use async at the end of the method. The method is just called getFavicon, not getfaviconasync.

Reply:
This C++ is from our COM API on top of which we generate a .NET wrapper and a WinRT wrapper. So its in our COM API without the Async suffix and our WinRT and .NET wrappers add the Async suffix to async methods. Its very specific to the tooling we have for WebView2.